db/compaction/compaction_job.cc

```diff
@@ -606,6 +606,10 @@ Status CompactionJob::Run() {
       status = state.status;
       io_s = state.io_status;
       break;
+    } else if (!state.io_status.ok()) {
```

Comment:
If `state.io_status` is not ok, `state.status` must be not ok. So this else branch should never be executed.